### PR TITLE
[WIP] feat: improve default address overrides

### PR DESCRIPTION
**Pull request by dryajov**
Improving `defaultAddrs` functionality. This should allow, overriding default API, Gateway and Swarm values separately.

### CODE REVIEW

File: src/factory-in-proc.js [hunk 1]
```diff
@@ -102,10 +102,19 @@ class FactoryInProc {
       ? options.init
       : true
 
-    if (options.disposable) {
-      options.config = defaults({}, options.config, defaultConfig)
-    } else {
-      const nonDisposableConfig = clone(defaultConfig)
+    const config = clone(defaultConfig)
+    if (options.defaultAddrs) {
+      if (options.config && options.config.Addresses) {
```
Comment (VictorBjelkholm) [inline src/factory-in-proc.js]: Seems this logic is duplicated in `src/factory-daemon.js `

Comment (dryajov) [reply src/factory-in-proc.js]: yes, we need a base class to compile all the duplicate logic into one place... I'll add it next.

**Comment by hugomrdias** [comment]
@dryajov can we get this on track ?

**Comment by hugomrdias** [comment]
doesn't apply anymore version 1.0.0 fixed this problem.

**Pull request closed without merging**